python/packages/aisearch/agent_framework_aisearch/_search_provider.py

```diff
@@ -0,0 +1,911 @@
+                Only needed when using API key authentication instead of managed identity.
+            knowledge_base_output_mode: Output mode for Knowledge Base retrieval. Only used in agentic mode.
+                "extractive_data": Returns raw chunks without synthesis (default, recommended for agent integration).
+                "answer_synthesis": Returns synthesized answer from the LLM.
+                Some knowledge sources require answer_synthesis mode. Default: "extractive_data".
+            retrieval_reasoning_effort: Reasoning effort for Knowledge Base query planning. Only used in agentic mode.
+                "minimal": Fastest, basic query planning.
+                "medium": Moderate reasoning with some query decomposition.
+                "low": Lower reasoning effort than medium.
+                Default: "minimal".
+            agentic_message_history_count: Number of recent messages from conversation history to send to
+                the Knowledge Base. This context helps with query planning in agentic mode, allowing the
+                Knowledge Base to understand the conversation flow and generate better retrieval queries.
+                There is no technical limit - adjust based on your use case. Default: 10.
+            env_file_path: Path to environment file for loading settings.
+            env_file_encoding: Encoding of the environment file.
+
+        Examples:
+            .. code-block:: python
+
+                from agent_framework_aisearch import AzureAISearchContextProvider
+                from azure.identity.aio import DefaultAzureCredential
+
+                # Using environment variables
+                # Set AZURE_SEARCH_ENDPOINT=https://mysearch.search.windows.net
+                # Set AZURE_SEARCH_INDEX_NAME=my-index
+                credential = DefaultAzureCredential()
+                provider = AzureAISearchContextProvider(credential=credential)
+
+                # Or passing parameters directly
+                provider = AzureAISearchContextProvider(
+                    endpoint="https://mysearch.search.windows.net",
+                    index_name="my-index",
+                    credential=credential,
+                )
+
+                # Or loading from a .env file
+                provider = AzureAISearchContextProvider(credential=credential, env_file_path="path/to/.env")
+        """
+        # Load settings from environment/file
+        try:
+            settings = AzureAISearchSettings(
+                endpoint=endpoint,
+                index_name=index_name,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as ex:
+            raise ServiceInitializationError("Failed to create Azure AI Search settings.", ex) from ex
+
+        # Use settings values, with explicit parameters taking precedence
+        endpoint = endpoint or settings.endpoint
```

Comment:
this isn;t needed, with the Settings class, the supplied arguments have precedence over the env variables, so if you pass in a value, that get's used, if you pass in nothing, it tries the env/env-file, so from the settings moving forward you can just refer to `settings.endpoint`